src/realm/tools/unicode_table_generator.cpp

```diff
@@ -0,0 +1,85 @@
+// This program generates the upper_lower array in unicode.cpp
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <vector>
+
+std::vector<std::string> tokenize(const char* src,
+    char delim,
+    bool want_empty_tokens = true)
+{
+    std::vector<std::string> tokens;
+
+    if (src && *src != '\0') // defensive
+        while (true) {
+            const char* d = strchr(src, delim);
+            size_t len = (d) ? d - src : strlen(src);
+
+            if (len || want_empty_tokens)
+                tokens.push_back(std::string(src, len)); // capture token
+
+            if (d)
+                src += len + 1;
+            else
+                break;
+        }
+    return tokens;
+}
+
+int hex2int(std::string hex)
+{
+    int x;
+    std::stringstream ss;
+    ss << std::hex << hex;
+    ss >> x;
+    return x;
+}
+
+int main()
+{
+    constexpr uint32_t last_unicode = 1023; // Last unicode we want to support. 1023 is the last greek unicode
+
+    // You need to download this file from unicode.org
+    std::ifstream infile("UnicodeData.txt");
+
+    std::string line;
+    int code = 0;
+
+    std::cout << "static const uint32_t upper_lower[" << last_unicode << " + 1][2] = {";

```

Comment:
Oh, true, I was too focused on this .cpp tool and forgot that you might mean the array definition in Core